testing/loki_alerts/run_single_test.sh

```diff
@@ -0,0 +1,222 @@
+            ;;
+        babysitter.log)
+            echo "in-erlang-multiline-log.conf"
+            ;;
+        *)
+            >&2 echo "ERROR: unknown fluent bit config for $1"
+            exit 1
+            ;;
+    esac
+}
+
+suite="$1"
+group=$(echo "$suite" | sed -e 's/\/.*//')
+alert=$(echo "$suite" | sed -e 's/.*\///')
+
+log 2 "Running suite: $suite"
+
+logs_path="$SCRIPT_DIR/$group/$alert"
+if [ ! -d "$logs_path" ]; then
+    echo "$group/$alert has no log files."
+    exit 1
+fi
+
+expr=$(yq e -e ".groups[] | select(.name == "'"'"$group"'"'") | .rules[] | select(.alert == "'"'"$alert"'"'") | .expr" "$SCRIPT_DIR/../../microlith/loki/alerting/couchbase/couchbase-rules.yaml")
+
+if [ -z "${LOKI_HOST:-}" ]; then
+    # Start up Loki
+    log 2 "Starting Loki..."
+    loki_container_id=$(docker run --rm -d -p 3100:3100 --name test_loki grafana/loki:2.4.1 -config.file=/etc/loki/local-config.yaml -log.level=debug)
+    LOKI_HOST="localhost"
+    LOKI_PORT=3100
+fi
+
+log 1 "Waiting for Loki to become ready..."
+wait_for_url 10 http://localhost:3100/ready
+log 2 "Loki ready."
+
+# Generate the Fluent Bit config - we can't just reuse it, because we need to set Read_from_Head or it'll skip straight to the end of the files.
+# Three main parts to it: a header and a trailer, and in between them a section for each config file we're reading]
+log 2 "Generating fluent-bit config..."
+fb_cfg=$(cat "$SCRIPT_DIR/fluent-bit.header.conf")
+
+for f in "$logs_path"/*; do
+    fb_cfg="$fb_cfg
+
+@include /fluent-bit/etc/couchbase/$(fluent_bit_config_for "$(basename "$f")")
+    Read_from_Head True
+"
+done
+
+fb_cfg="$fb_cfg
+$(cat "$SCRIPT_DIR/fluent-bit.trailer.conf")
+"
+
+echo "$fb_cfg" > "$SCRIPT_DIR/fluent-bit.conf"
+log 2 "Fluent Bit config generated and written to $SCRIPT_DIR/fluent-bit.conf"
+
+# Run Fluent Bit over the case's files
+# Would like to use Exit_on_EOF, but that'd require setting it for each individual [INPUT], which is infeasible (also https://github.com/fluent/fluent-bit/issues/3274)
+# Instead we just run FB and wait a bit - long enough for it to dump the full logs under any reasonable circumstances
+log 1 "Starting Fluent Bit..."
+fb_command="docker run --rm -d \
+    --network host \
+    -v "'"'"$logs_path:/opt/couchbase/var/lib/couchbase/logs:ro"'"'" \
+    -v "'"'"$SCRIPT_DIR/fluent-bit.conf:/fluent-bit/config/fluent-bit.conf"'"'" \
+    -v "'"'"$SCRIPT_DIR/rebase_times.lua:/fluent-bit/config/rebase_times.lua"'"'" \
+    -e 'LOKI_MATCH=*' \
+    -e 'LOKI_HOST=$LOKI_HOST' \
+    -e 'LOKI_PORT=$LOKI_PORT' \
+    -e 'LOKI_TENANT=$suite' \
+    --name test_fluentbit \
+    couchbase/fluent-bit:1.1.3"
+log 2 "Command: $fb_command"
+fb_container_id=$(eval "$fb_command")
+
+# Print the logs so we see what's going on.
+if [ "$VERBOSITY" -lt 2 ]; then
+    sleep 10
+else
+    # timeout will abort the script with error 124 without the ||true
+    timeout 10 docker logs -ft "$fb_container_id" || true
```

Comment:
Could be worth documenting the wait time being 10 sec / changing these to be a variable declared at line 157